src/devices/vdml.cpp

```diff
@@ -0,0 +1,35 @@
+#include "vdml/vdml.hpp"
+
+#include "vdml/registry.hpp"
+
+#include <cstdlib>
+#include <mutex>
+
+using namespace zest::vdml;
+
+// initialize the device mutex array
+std::array<std::mutex, 32> device_mutexes;
```

Comment:
This can be `constinit`
```suggestion
constinit std::array<std::mutex, 32> device_mutexes;
```